pex/cli/commands/lock.py

```diff
@@ -220,3 +277,149 @@ def _export(self):
                 ),
             )
         )
+
+    def _update(self):
+        # type: () -> Result
+        lockfile_path = self.options.lockfile[0]
+        lock_file = try_(self._load_lockfile(lockfile_path=lockfile_path))
+
+        target_configuration = target_options.configure(self.options)
+        targets_by_platform_tag = OrderedDict(
+            (target.get_supported_tags()[0], target)
+            for target in target_configuration.unique_targets()
+        )  # type: OrderedDict[tags.Tag, DistributionTarget]
+
+        if self.options.strict:
+            required_platforms = {
+                locked_resolve.platform_tag for locked_resolve in lock_file.locked_resolves
+            }
+            platforms_in_hand = required_platforms.intersection(targets_by_platform_tag.keys())
+            if platforms_in_hand != required_platforms:
+                return Error(
+                    "This lock update is --strict but the following platforms present in "
+                    "{lock_file} were not found on the local machine:\n"
+                    "{missing_platforms}\n"
+                    "You might be able to correct this by adjusting target options like "
+                    "--python-path or else by relaxing the update to be --non-strict.".format(
+                        lock_file=lockfile_path,
+                        missing_platforms="\n".join(
+                            sorted(
+                                "+ {platform}".format(platform=platform)
+                                for platform in required_platforms - platforms_in_hand
+                            )
+                        ),
+                    )
+                )
+
+        targets_to_update = OrderedDict()  # type: OrderedDict[DistributionTarget, LockedResolve]
+        for locked_resolve in lock_file.locked_resolves:
+            target = targets_by_platform_tag.get(locked_resolve.platform_tag)
+            if target:
+                targets_to_update[target] = locked_resolve
+
+        if not targets_to_update:
+            return Ok()
```

Comment:
Should this fail instead with "no targets matched"?

Reply:
Perhaps. Succeed or fail though, a message calling the no-op out is definitely needed.

Reply:
Kept this Ok, but added a message indicating no lock update was performed and how to possibly change this by tweaking --python-path.